src/edu/csus/ecs/pc2/clics/API202306/CLICSContestState.java

```diff
@@ -0,0 +1,149 @@
+            if (model.getContestTime().isPastEndOfContest()) {
+                Calendar endedDate = IJSONTool.calculateElapsedWalltime(model, ct.getContestLengthMS());
+                if (endedDate != null) {
+                    ended = Utilities.getIso8601formatterWithMS().format(endedDate.getTimeInMillis());
+                }
+            }
+
+            contest_time = ContestTime.formatTimeMS(ct.getElapsedMS());
+            if(ct.isContestRunning() == false) {
+                if(ended == null) {
+                    // This means the contest was paused - can't pause a contest if it is ended... so ended must be null
+                    paused = contest_time;
+                }
+            }
+            resumed = Utilities.getIso8601formatterWithMS().format(ct.getResumeTime().getTime());
+
+            String scoreboardFreezeDuration = ci.getFreezeTime();
+            Date thawedDate = null;
+
+            if (scoreboardFreezeDuration != null && scoreboardFreezeDuration.trim().length() > 0) {
+                long elapsed = ct.getElapsedSecs();
+                long freezeTimeSecs = Utilities.getFreezeTime(model);
+                // FIXME this date should be stored in ContestInformation
+                if (elapsed >= freezeTimeSecs) {
+                    Calendar freezeCal = IJSONTool.calculateElapsedWalltime(model, freezeTimeSecs * 1000);
+                    if (freezeCal != null) {
+                        frozen = Utilities.getIso8601formatterWithMS().format(freezeCal.getTime());
+                    }
+                }
+                if (ci.isUnfrozen()) {
+                    thawedDate = model.getContestInformation().getThawed();
+                    if (thawedDate != null) {
+                        thawed = Utilities.getIso8601formatterWithMS().format(thawedDate);
+                    }
+                }
+            }
+            // FIXME this should only be showed if the contest is thawed for public users
+            FinalizeData fData = model.getFinalizeData();
+            if (fData != null) {
+                Date fDate = fData.getCertificationDate();
+                if(fDate != null) {
+                    finalized = Utilities.getIso8601formatterWithMS().format(fDate);
+                    // If contest was thawed after finalized, then use thaw time
+                    if(thawed != null && thawedDate != null && thawedDate.after(fDate)) {
+                        end_of_updates = thawed;
+                    } else {
+                        end_of_updates = finalized;
+                    }
+                }
+            }
+        }
+    }
+
+    public String toJSON() {
+
+        try {
+            ObjectMapper mapper = JSONUtilities.getObjectMapper();
+            return mapper.writeValueAsString(this);
+        } catch (Exception e) {
+            return "Error creating JSON for contest state info " + e.getMessage();
```

Comment:
I would suggest this error message explicitly mention `...for CLICS Contest State info`... just to make it clear that the Exception is related to the CLICS class and not a PC2 class.